lib/Transport.js

```diff
@@ -494,6 +536,59 @@ class Transport {
       callback(null, hosts)
     })
   }
+
+  productCheck () {
+    debug('Start product check')
+    this[kProductCheck] = 1
+    this.request({
+      method: 'GET',
+      path: '/'
+    }, (err, result) => {
+      this[kProductCheck] = 3
+      if (err) {
+        debug('Product check failed', err)
+        if (err.statusCode === 401 || err.statusCode === 403) {
+          this[kProductCheck] = 2
+          process.emitWarning('The client is unable to verify that the server is Elasticsearch due to security privileges on the server side. Some functionality may not be compatible if the server is running an unsupported product.')
+          productCheckEmitter.emit('product-check', true)
+        } else {
+          this[kProductCheck] = 0
+          productCheckEmitter.emit('product-check', false)
+        }
+      } else {
+        debug('Checking elasticsearch version', result.body, result.headers)
+        if (result.body.version == null || typeof result.body.version.number !== 'string') {
+          debug('Can\'t access Elasticsearch version')
+          return productCheckEmitter.emit('product-check', false)
+        }
+        const tagline = result.body.tagline
+        const version = result.body.version.number.split('.')
+        const major = Number(version[0])
+        const minor = Number(version[1])
+        if (major < 6) {
+          return productCheckEmitter.emit('product-check', false)
+        } else if (major >= 6 && major < 7) {
+          if (tagline !== 'You Know, for Search') {
+            debug('Bad tagline')
+            return productCheckEmitter.emit('product-check', false)
+          }
+        } else if (major === 7 && minor < 14) {
+          if (tagline !== 'You Know, for Search' || result.body.version.build_flavor !== 'default') {
```

Comment:
FYI this breaks support for the oss docker images <= 7.10.1.
The build_flavor in there s `oss`.
```
{
  "name" : "elasticsearch-production-2",
  "cluster_name" : "elasticsearch-bmg-prd",
  "cluster_uuid" : "41tljEp7T66B-13N2zaE4A",
  "version" : {
    "number" : "7.10.1",
    "build_flavor" : "oss",
    "build_type" : "docker",
    "build_hash" : "1c34507e66d7db1211f66f3513706fdf548736aa",
    "build_date" : "2020-12-05T01:00:33.671820Z",
    "build_snapshot" : false,
    "lucene_version" : "8.7.0",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"
}
```

While an upgrade to a newer image fixes the incompatibility, you might want to adopt the check in here.

Reply:
Hello! Please see https://github.com/elastic/elasticsearch-js/issues/1519.

Reply:
Thanks Thomas, that's then on purpose as I've assumed. There's just no mention in this PR, which could have the change a bit more transparent.